subsys/net/ip/ipv4_autoconf.c

```diff
@@ -0,0 +1,270 @@
+/** @file
+ * @brief IPv4 autoconf related functions
+ */
+
+/*
+ * Copyright (c) 2017 Matthias Boesl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_IPV4_AUTOCONF)
+#define SYS_LOG_DOMAIN "net/ipv4ll"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include "net_private.h"
+#include <errno.h>
+#include <net/arp.h>
+#include <net/net_pkt.h>
+#include <net/net_core.h>
+#include <net/net_if.h>
+
+#include <net/ipv4_autoconf.h>
+
+#define BUF_ALLOC_TIMEOUT K_MSEC(100)
+
+static void send_probe(struct net_if *iface);
+static void send_announcement(struct net_if *iface);
+
+static inline struct net_pkt *prepare_arp(struct net_if *iface)
+{
+	struct net_pkt *pkt;
+	struct net_buf *frag;
+	struct net_arp_hdr *hdr;
+	struct net_eth_hdr *eth;
+
+	pkt = net_pkt_get_reserve_tx(sizeof(struct net_eth_hdr), BUF_ALLOC_TIMEOUT);
+	if (!pkt) {
+		goto fail;
+	}
+
+	frag = net_pkt_get_frag(pkt, K_FOREVER);
+	if (!frag) {
+		goto fail;
+	}
+
+	net_pkt_frag_add(pkt, frag);
+	net_pkt_set_iface(pkt, iface);
+	net_pkt_set_family(pkt, AF_INET);
+	net_pkt_set_ll_reserve(pkt, sizeof(struct net_eth_hdr));
+
+	eth = NET_ETH_HDR(pkt);
+	hdr = NET_ARP_HDR(pkt);
+	eth->type = htons(NET_ETH_PTYPE_ARP);
+
+	/* ethernet header */
+	memcpy(&eth->src.addr, net_if_get_link_addr(iface)->addr,
+	       sizeof(struct net_eth_addr));
+	memset(&eth->dst.addr, 0xFF, sizeof(struct net_eth_addr));
+
+	hdr->hwtype = htons(NET_ARP_HTYPE_ETH);
+	hdr->protocol = htons(NET_ETH_PTYPE_IP);
+	hdr->hwlen = sizeof(struct net_eth_addr);
+	hdr->protolen = sizeof(struct in_addr);
+	hdr->opcode = htons(NET_ARP_REQUEST);
+
+	memcpy(&hdr->src_hwaddr.addr, &eth->src.addr,
+	       sizeof(struct net_eth_addr));
+	memset(&hdr->dst_hwaddr.addr, 0x00, sizeof(struct net_eth_addr));
+
+	net_ipaddr_copy(&hdr->dst_ipaddr, &net_if_get_config(iface)->ipv4auto.current_ip);
```

Comment:
too long lines here, I suggest you use helper to get a config and then use the config var so that the lines will fit 80 chars